Helm lint throws an error where a warning should be thrown instead

As a result of these 2 PRs:

1. **https://github.com/helm/helm/pull/7986 - feat: implement deprecation warnings in helm lint**
1. **https://github.com/helm/helm/pull/8011 - feat: lint the names of templated resources**

now `helm lint` validates  whether resource `metadata.name` complies Kubernetes naming rules and whether the `apiVersion` is up to date. And these are great features, thank you for implementing them! 

However, these rules are having ERROR severity instead of WARNING. These should be warnings because these are NOT errors from the Kubernetes perspective - Kubernetes API still accepts resources with deprecated API versions as well as resources with metadata names that, for example, contain `:` in them. If these are warnings instead of errors, then it will be up to a client to decide whether he wants to fail lint (via using `--strict` flag for `helm lint` command) or not.

Setting the error severity level for these issues also complicates the migration of automated deployments 
 to helm 3.3, especially form Helm 2.x.

_How to fix?_ - I believe [these lines](https://github.com/technosophos/k8s-helm/blob/2d1ce6d7866e1edc2431eb637864f0bdc7e6a0a7/pkg/lint/rules/template.go#L133-L134) of code should be rewritten to:

```go
linter.RunLinterRule(support.WarningSev, path, validateMetadataName(&yamlStruct))
linter.RunLinterRule(support.WarningSev, path, validateNoDeprecations(&yamlStruct))
```

@bacongobbler, maybe it is true for some APIs but certainly not for all - here is an example... The helm chart has a `ClusterRole` resource defined using the deprecated API version:

```yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  name: {{ include "example-chart.fullname" . }}
  labels: {{ include "example-chart.labels" . | nindent 4 }}
rules:
  {{- toYaml .Values.clusterRoleRules | nindent 2 }}
```

If you run an installation/upgrade of this chart, it will succeed and there will be no errors. However, `helm lint` command in Helm 3.3 fails for it with the error:

```
[ERROR] templates/clusterrole.yaml: the kind "rbac.authorization.k8s.io/v1beta1 ClusterRole" is deprecated in favor of "rbac.authorization.k8s.io/v1 ClusterRole"
```

The same applies to 
**https://github.com/helm/helm/pull/8011 - feat: lint the names of templated resources**
For example, [prometheus-adapter](https://github.com/helm/charts/tree/master/stable/prometheus-adapter) gets successfully installed/upgraded, but `helm lint` command in Helm 3.3 fails for it with the error:
```
[ERROR] templates/custom-metrics-apiserver-auth-delegator-cluster-role-binding.yaml: object name does not conform to Kubernetes naming requirements: "prometheus-adapter:system:auth-delegator"
```

Yes, and here is another example...
The helm chart has an `Ingress` resource defined using the deprecated API version:

```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  # ...
spec:
 # ...
```

If you run an installation/upgrade of this chart, it will succeed (the cluster's Kubernetes version is **1.18.7**) and there will be no errors. However, `helm lint` command in Helm 3.3 fails for it with the error:

```
[ERROR] templates/ingress.yaml: the kind "extensions/v1beta1 Ingress" is deprecated in favor of "networking.k8s.io/v1beta1 Ingress"
```

--- a/Makefile
+++ b/Makefile
@@ -56,8 +56,9 @@ LDFLAGS += -X helm.sh/helm/v3/internal/version.gitTreeState=${GIT_DIRTY}
 LDFLAGS += $(EXT_LDFLAGS)
 
 # Define constants based on the client-go version
-K8S_MODULES_MAJOR_VER=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[1]+1}' go.mod)
-K8S_MODULES_MINOR_VER=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[2]}' go.mod)
+K8S_MODULES_VER=$(subst ., ,$(subst v,,$(shell go list -f '{{.Version}}' -m k8s.io/client-go)))
+K8S_MODULES_MAJOR_VER=$(shell echo $$(($(firstword $(K8S_MODULES_VER)) + 1)))
+K8S_MODULES_MINOR_VER=$(word 2,$(K8S_MODULES_VER))
 
 LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMajor=$(K8S_MODULES_MAJOR_VER)
 LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMinor=$(K8S_MODULES_MINOR_VER)

--- a/Makefile
+++ b/Makefile
@@ -55,6 +55,10 @@ LDFLAGS += -X helm.sh/helm/v3/internal/version.gitCommit=${GIT_COMMIT}
 LDFLAGS += -X helm.sh/helm/v3/internal/version.gitTreeState=${GIT_DIRTY}
 LDFLAGS += $(EXT_LDFLAGS)
 
+# Define constants for the client-go version
+LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMajor=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[1]+1}' go.mod)
+LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMinor=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[2]}' go.mod)
+
 .PHONY: all
 all: build
 

--- a/Makefile
+++ b/Makefile
@@ -55,9 +55,14 @@ LDFLAGS += -X helm.sh/helm/v3/internal/version.gitCommit=${GIT_COMMIT}
 LDFLAGS += -X helm.sh/helm/v3/internal/version.gitTreeState=${GIT_DIRTY}
 LDFLAGS += $(EXT_LDFLAGS)
 
-# Define constants for the client-go version
-LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMajor=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[1]+1}' go.mod)
-LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMinor=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[2]}' go.mod)
+# Define constants based on the client-go version
+K8S_MODULES_MAJOR_VER=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[1]+1}' go.mod)
+K8S_MODULES_MINOR_VER=$(shell awk '/client-go/{s=$$2;sub("^v","",s);split(s,v,".");print v[2]}' go.mod)
+
+LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMajor=$(K8S_MODULES_MAJOR_VER)
+LDFLAGS += -X helm.sh/helm/v3/pkg/lint/rules.k8sVersionMinor=$(K8S_MODULES_MINOR_VER)
+LDFLAGS += -X helm.sh/helm/v3/pkg/chartutil.k8sVersionMajor=$(K8S_MODULES_MAJOR_VER)
+LDFLAGS += -X helm.sh/helm/v3/pkg/chartutil.k8sVersionMinor=$(K8S_MODULES_MINOR_VER)
 
 .PHONY: all
 all: build

--- a/cmd/helm/testdata/output/template-name-template.txt
+++ b/cmd/helm/testdata/output/template-name-template.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "foobar-YWJj-baz"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
 spec:
   type: ClusterIP
   ports:

--- a/cmd/helm/testdata/output/template-name-template.txt
+++ b/cmd/helm/testdata/output/template-name-template.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "foobar-YWJj-baz"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
 spec:
   type: ClusterIP
   ports:

--- a/cmd/helm/testdata/output/template-set.txt
+++ b/cmd/helm/testdata/output/template-set.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
 spec:
   type: ClusterIP
   ports:

--- a/cmd/helm/testdata/output/template-set.txt
+++ b/cmd/helm/testdata/output/template-set.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
 spec:
   type: ClusterIP
   ports:

--- a/cmd/helm/testdata/output/template-show-only-multiple.txt
+++ b/cmd/helm/testdata/output/template-show-only-multiple.txt
@@ -8,8 +8,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-show-only-multiple.txt
+++ b/cmd/helm/testdata/output/template-show-only-multiple.txt
@@ -8,8 +8,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-show-only-one.txt
+++ b/cmd/helm/testdata/output/template-show-only-one.txt
@@ -8,8 +8,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-show-only-one.txt
+++ b/cmd/helm/testdata/output/template-show-only-one.txt
@@ -8,8 +8,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-skip-tests.txt
+++ b/cmd/helm/testdata/output/template-skip-tests.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-values-files.txt
+++ b/cmd/helm/testdata/output/template-values-files.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
 spec:
   type: ClusterIP
   ports:

--- a/cmd/helm/testdata/output/template-values-files.txt
+++ b/cmd/helm/testdata/output/template-values-files.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
 spec:
   type: ClusterIP
   ports:

--- a/cmd/helm/testdata/output/template-with-api-version.txt
+++ b/cmd/helm/testdata/output/template-with-api-version.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-with-api-version.txt
+++ b/cmd/helm/testdata/output/template-with-api-version.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-with-crds.txt
+++ b/cmd/helm/testdata/output/template-with-crds.txt
@@ -89,8 +89,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template-with-crds.txt
+++ b/cmd/helm/testdata/output/template-with-crds.txt
@@ -89,8 +89,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
     kube-api-version/test: v1
 spec:
   type: ClusterIP

--- a/cmd/helm/testdata/output/template.txt
+++ b/cmd/helm/testdata/output/template.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "19"
-    kube-version/version: "v1.19.0"
+    kube-version/minor: "20"
+    kube-version/version: "v1.20.0"
 spec:
   type: ClusterIP
   ports:

--- a/cmd/helm/testdata/output/template.txt
+++ b/cmd/helm/testdata/output/template.txt
@@ -72,8 +72,8 @@ metadata:
     helm.sh/chart: "subchart-0.1.0"
     app.kubernetes.io/instance: "RELEASE-NAME"
     kube-version/major: "1"
-    kube-version/minor: "18"
-    kube-version/version: "v1.18.0"
+    kube-version/minor: "19"
+    kube-version/version: "v1.19.0"
 spec:
   type: ClusterIP
   ports:

--- a/pkg/chartutil/capabilities.go
+++ b/pkg/chartutil/capabilities.go
@@ -29,7 +29,7 @@ import (
 
 const (
 	k8sVersionMajor = 1
-	k8sVersionMinor = 19
+	k8sVersionMinor = 20
 )
 
 var (

--- a/pkg/chartutil/capabilities.go
+++ b/pkg/chartutil/capabilities.go
@@ -16,12 +16,20 @@ limitations under the License.
 package chartutil
 
 import (
+	"fmt"
+	"strconv"
+
 	"k8s.io/client-go/kubernetes/scheme"
 
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 
 	helmversion "helm.sh/helm/v3/internal/version"
+)
+
+const (
+	k8sMajorVer = 1
+	k8sMinorVer = 19
 )
 
 var (
@@ -31,9 +39,9 @@ var (
 	// DefaultCapabilities is the default set of capabilities.
 	DefaultCapabilities = &Capabilities{
 		KubeVersion: KubeVersion{
-			Version: "v1.18.0",
-			Major:   "1",
-			Minor:   "18",
+			Version: fmt.Sprintf("v%d.%d.0", k8sMajorVer, k8sMinorVer),
+			Major:   strconv.Itoa(k8sMajorVer),
+			Minor:   strconv.Itoa(k8sMinorVer),
 		},
 		APIVersions: DefaultVersionSet,
 		HelmVersion: helmversion.Get(),

--- a/pkg/chartutil/capabilities.go
+++ b/pkg/chartutil/capabilities.go
@@ -28,8 +28,8 @@ import (
 )
 
 const (
-	k8sMajorVer = 1
-	k8sMinorVer = 19
+	k8sVersionMajor = 1
+	k8sVersionMinor = 19
 )
 
 var (
@@ -39,9 +39,9 @@ var (
 	// DefaultCapabilities is the default set of capabilities.
 	DefaultCapabilities = &Capabilities{
 		KubeVersion: KubeVersion{
-			Version: fmt.Sprintf("v%d.%d.0", k8sMajorVer, k8sMinorVer),
-			Major:   strconv.Itoa(k8sMajorVer),
-			Minor:   strconv.Itoa(k8sMinorVer),
+			Version: fmt.Sprintf("v%d.%d.0", k8sVersionMajor, k8sVersionMinor),
+			Major:   strconv.Itoa(k8sVersionMajor),
+			Minor:   strconv.Itoa(k8sVersionMinor),
 		},
 		APIVersions: DefaultVersionSet,
 		HelmVersion: helmversion.Get(),

--- a/pkg/chartutil/capabilities_test.go
+++ b/pkg/chartutil/capabilities_test.go
@@ -42,20 +42,20 @@ func TestDefaultVersionSet(t *testing.T) {
 
 func TestDefaultCapabilities(t *testing.T) {
 	kv := DefaultCapabilities.KubeVersion
-	if kv.String() != "v1.19.0" {
-		t.Errorf("Expected default KubeVersion.String() to be v1.19.0, got %q", kv.String())
+	if kv.String() != "v1.20.0" {
+		t.Errorf("Expected default KubeVersion.String() to be v1.20.0, got %q", kv.String())
 	}
-	if kv.Version != "v1.19.0" {
-		t.Errorf("Expected default KubeVersion.Version to be v1.19.0, got %q", kv.Version)
+	if kv.Version != "v1.20.0" {
+		t.Errorf("Expected default KubeVersion.Version to be v1.20.0, got %q", kv.Version)
 	}
-	if kv.GitVersion() != "v1.19.0" {
-		t.Errorf("Expected default KubeVersion.GitVersion() to be v1.19.0, got %q", kv.Version)
+	if kv.GitVersion() != "v1.20.0" {
+		t.Errorf("Expected default KubeVersion.GitVersion() to be v1.20.0, got %q", kv.Version)
 	}
 	if kv.Major != "1" {
 		t.Errorf("Expected default KubeVersion.Major to be 1, got %q", kv.Major)
 	}
-	if kv.Minor != "19" {
-		t.Errorf("Expected default KubeVersion.Minor to be 19, got %q", kv.Minor)
+	if kv.Minor != "20" {
+		t.Errorf("Expected default KubeVersion.Minor to be 20, got %q", kv.Minor)
 	}
 }
 

--- a/pkg/chartutil/capabilities_test.go
+++ b/pkg/chartutil/capabilities_test.go
@@ -42,20 +42,20 @@ func TestDefaultVersionSet(t *testing.T) {
 
 func TestDefaultCapabilities(t *testing.T) {
 	kv := DefaultCapabilities.KubeVersion
-	if kv.String() != "v1.18.0" {
-		t.Errorf("Expected default KubeVersion.String() to be v1.18.0, got %q", kv.String())
+	if kv.String() != "v1.19.0" {
+		t.Errorf("Expected default KubeVersion.String() to be v1.19.0, got %q", kv.String())
 	}
-	if kv.Version != "v1.18.0" {
-		t.Errorf("Expected default KubeVersion.Version to be v1.18.0, got %q", kv.Version)
+	if kv.Version != "v1.19.0" {
+		t.Errorf("Expected default KubeVersion.Version to be v1.19.0, got %q", kv.Version)
 	}
-	if kv.GitVersion() != "v1.18.0" {
-		t.Errorf("Expected default KubeVersion.GitVersion() to be v1.18.0, got %q", kv.Version)
+	if kv.GitVersion() != "v1.19.0" {
+		t.Errorf("Expected default KubeVersion.GitVersion() to be v1.19.0, got %q", kv.Version)
 	}
 	if kv.Major != "1" {
 		t.Errorf("Expected default KubeVersion.Major to be 1, got %q", kv.Major)
 	}
-	if kv.Minor != "18" {
-		t.Errorf("Expected default KubeVersion.Minor to be 18, got %q", kv.Minor)
+	if kv.Minor != "19" {
+		t.Errorf("Expected default KubeVersion.Minor to be 19, got %q", kv.Minor)
 	}
 }
 

--- a/pkg/lint/rules/deprecations.go
+++ b/pkg/lint/rules/deprecations.go
@@ -29,7 +29,7 @@ const (
 	// This should be set in the Makefile based on the version of client-go being imported.
 	// These constants will be overwritten with LDFLAGS
 	k8sVersionMajor = 1
-	k8sVersionMinor = 19
+	k8sVersionMinor = 20
 )
 
 // deprecatedAPIError indicates than an API is deprecated in Kubernetes

--- a/pkg/lint/rules/deprecations.go
+++ b/pkg/lint/rules/deprecations.go
@@ -18,50 +18,176 @@ package rules // import "helm.sh/helm/v3/pkg/lint/rules"
 
 import "fmt"
 
-// deprecatedAPIs lists APIs that are deprecated (left) with suggested alternatives (right).
-//
-// An empty rvalue indicates that the API is completely deprecated.
-var deprecatedAPIs = map[string]string{
-	"extensions/v1beta1 Deployment":                             "apps/v1 Deployment",
-	"extensions/v1beta1 DaemonSet":                              "apps/v1 DaemonSet",
-	"extensions/v1beta1 ReplicaSet":                             "apps/v1 ReplicaSet",
-	"extensions/v1beta1 PodSecurityPolicy":                      "policy/v1beta1 PodSecurityPolicy",
-	"extensions/v1beta1 NetworkPolicy":                          "networking.k8s.io/v1beta1 NetworkPolicy",
-	"extensions/v1beta1 Ingress":                                "networking.k8s.io/v1beta1 Ingress",
-	"apps/v1beta1 Deployment":                                   "apps/v1 Deployment",
-	"apps/v1beta1 StatefulSet":                                  "apps/v1 StatefulSet",
-	"apps/v1beta1 ReplicaSet":                                   "apps/v1 ReplicaSet",
-	"apps/v1beta2 Deployment":                                   "apps/v1 Deployment",
-	"apps/v1beta2 StatefulSet":                                  "apps/v1 StatefulSet",
-	"apps/v1beta2 DaemonSet":                                    "apps/v1 DaemonSet",
-	"apps/v1beta2 ReplicaSet":                                   "apps/v1 ReplicaSet",
-	"apiextensions.k8s.io/v1beta1 CustomResourceDefinition":     "apiextensions.k8s.io/v1 CustomResourceDefinition",
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRole":            "rbac.authorization.k8s.io/v1 ClusterRole",
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleList":        "rbac.authorization.k8s.io/v1 ClusterRoleList",
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleBinding":     "rbac.authorization.k8s.io/v1 ClusterRoleBinding",
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleBindingList": "rbac.authorization.k8s.io/v1 ClusterRoleBindingList",
-	"rbac.authorization.k8s.io/v1alpha1 Role":                   "rbac.authorization.k8s.io/v1 Role",
-	"rbac.authorization.k8s.io/v1alpha1 RoleList":               "rbac.authorization.k8s.io/v1 RoleList",
-	"rbac.authorization.k8s.io/v1alpha1 RoleBinding":            "rbac.authorization.k8s.io/v1 RoleBinding",
-	"rbac.authorization.k8s.io/v1alpha1 RoleBindingList":        "rbac.authorization.k8s.io/v1 RoleBindingList",
-	"rbac.authorization.k8s.io/v1beta1 ClusterRole":             "rbac.authorization.k8s.io/v1 ClusterRole",
-	"rbac.authorization.k8s.io/v1beta1 ClusterRoleList":         "rbac.authorization.k8s.io/v1 ClusterRoleList",
-	"rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding":      "rbac.authorization.k8s.io/v1 ClusterRoleBinding",
-	"rbac.authorization.k8s.io/v1beta1 ClusterRoleBindingList":  "rbac.authorization.k8s.io/v1 ClusterRoleBindingList",
-	"rbac.authorization.k8s.io/v1beta1 Role":                    "rbac.authorization.k8s.io/v1 Role",
-	"rbac.authorization.k8s.io/v1beta1 RoleList":                "rbac.authorization.k8s.io/v1 RoleList",
-	"rbac.authorization.k8s.io/v1beta1 RoleBinding":             "rbac.authorization.k8s.io/v1 RoleBinding",
-	"rbac.authorization.k8s.io/v1beta1 RoleBindingList":         "rbac.authorization.k8s.io/v1 RoleBindingList",
+type deprecatedAPI struct {
+	NewAPI       string
+	DeprecatedIn string
+	RemovedIn    string
+}
+
+// deprecatedAPIs lists APIs that are deprecated (key) with suggested alternatives (value).
+var deprecatedAPIs = map[string]deprecatedAPI{
+	"extensions/v1beta1 Deployment": {
+		NewAPI:       "apps/v1 Deployment",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"apps/v1beta1 Deployment": {
+		NewAPI:       "apps/v1 Deployment",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"apps/v1beta2 Deployment": {
+		NewAPI:       "apps/v1 Deployment",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"apps/v1beta1 StatefulSet": {
+		NewAPI:       "apps/v1 StatefulSet",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"apps/v1beta2 StatefulSet": {
+		NewAPI:       "apps/v1 StatefulSet",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"extensions/v1beta1 DaemonSet": {
+		NewAPI:       "apps/v1 DaemonSet",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"apps/v1beta2 DaemonSet": {
+		NewAPI:       "apps/v1 DaemonSet",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"extensions/v1beta1 ReplicaSet": {
+		NewAPI:       "apps/v1 ReplicaSet",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"apps/v1beta1 ReplicaSet": {
+		NewAPI:       "apps/v1 ReplicaSet",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"apps/v1beta2 ReplicaSet": {
+		NewAPI:       "apps/v1 ReplicaSet",
+		DeprecatedIn: "v1.9",
+		RemovedIn:    "v1.16",
+	},
+	"extensions/v1beta1 NetworkPolicy": {
+		NewAPI:       "networking.k8s.io/v1 NetworkPolicy",
+		DeprecatedIn: "v1.8",
+		RemovedIn:    "v1.16",
+	},
+	"extensions/v1beta1 PodSecurityPolicy": {
+		NewAPI:       "policy/v1beta1 PodSecurityPolicy",
+		DeprecatedIn: "v1.10",
+		RemovedIn:    "v1.16",
+	},
+	"apiextensions.k8s.io/v1beta1 CustomResourceDefinition": {
+		NewAPI:       "apiextensions.k8s.io/v1 CustomResourceDefinition",
+		DeprecatedIn: "v1.16",
+		RemovedIn:    "v1.19",
+	},
+	"extensions/v1beta1 Ingress": {
+		NewAPI:       "networking.k8s.io/v1beta1 Ingress",
+		DeprecatedIn: "v1.14",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 ClusterRole": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRole",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleBinding": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBinding",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleBindingList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBindingList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 Role": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 Role",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 RoleList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 RoleList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 RoleBinding": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBinding",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1alpha1 RoleBindingList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBindingList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta ClusterRole": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRole",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta1 ClusterRoleList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBinding",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta1 ClusterRoleBindingList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBindingList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta1 Role": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 Role",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta1 RoleList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 RoleList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta1 RoleBinding": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBinding",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
+	"rbac.authorization.k8s.io/v1beta1 RoleBindingList": {
+		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBindingList",
+		DeprecatedIn: "v1.17",
+		RemovedIn:    "v1.22",
+	},
 }
 
 // deprecatedAPIError indicates than an API is deprecated in Kubernetes
 type deprecatedAPIError struct {
 	Deprecated  string
 	Alternative string
+	In          string
+	Removed     string
 }
 
 func (e deprecatedAPIError) Error() string {
-	msg := fmt.Sprintf("the kind %q is deprecated", e.Deprecated)
+	msg := fmt.Sprintf("the kind %q is deprecated as of %q and removed in %q", e.Deprecated, e.In, e.Removed)
 	if e.Alternative != "" {
 		msg += fmt.Sprintf(" in favor of %q", e.Alternative)
 	}
@@ -70,10 +196,12 @@ func (e deprecatedAPIError) Error() string {
 
 func validateNoDeprecations(resource *K8sYamlStruct) error {
 	gvk := fmt.Sprintf("%s %s", resource.APIVersion, resource.Kind)
-	if alt, ok := deprecatedAPIs[gvk]; ok {
+	if dep, ok := deprecatedAPIs[gvk]; ok {
 		return deprecatedAPIError{
 			Deprecated:  gvk,
-			Alternative: alt,
+			Alternative: dep.NewAPI,
+			In:          dep.DeprecatedIn,
+			Removed:     dep.RemovedIn,
 		}
 	}
 	return nil

--- a/pkg/lint/rules/deprecations.go
+++ b/pkg/lint/rules/deprecations.go
@@ -16,193 +16,64 @@ limitations under the License.
 
 package rules // import "helm.sh/helm/v3/pkg/lint/rules"
 
-import "fmt"
+import (
+	"fmt"
 
-type deprecatedAPI struct {
-	NewAPI       string
-	DeprecatedIn string
-	RemovedIn    string
-}
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/deprecation"
+	kscheme "k8s.io/client-go/kubernetes/scheme"
+)
 
-// deprecatedAPIs lists APIs that are deprecated (key) with suggested alternatives (value).
-var deprecatedAPIs = map[string]deprecatedAPI{
-	"extensions/v1beta1 Deployment": {
-		NewAPI:       "apps/v1 Deployment",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"apps/v1beta1 Deployment": {
-		NewAPI:       "apps/v1 Deployment",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"apps/v1beta2 Deployment": {
-		NewAPI:       "apps/v1 Deployment",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"apps/v1beta1 StatefulSet": {
-		NewAPI:       "apps/v1 StatefulSet",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"apps/v1beta2 StatefulSet": {
-		NewAPI:       "apps/v1 StatefulSet",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"extensions/v1beta1 DaemonSet": {
-		NewAPI:       "apps/v1 DaemonSet",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"apps/v1beta2 DaemonSet": {
-		NewAPI:       "apps/v1 DaemonSet",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"extensions/v1beta1 ReplicaSet": {
-		NewAPI:       "apps/v1 ReplicaSet",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"apps/v1beta1 ReplicaSet": {
-		NewAPI:       "apps/v1 ReplicaSet",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"apps/v1beta2 ReplicaSet": {
-		NewAPI:       "apps/v1 ReplicaSet",
-		DeprecatedIn: "v1.9",
-		RemovedIn:    "v1.16",
-	},
-	"extensions/v1beta1 NetworkPolicy": {
-		NewAPI:       "networking.k8s.io/v1 NetworkPolicy",
-		DeprecatedIn: "v1.8",
-		RemovedIn:    "v1.16",
-	},
-	"extensions/v1beta1 PodSecurityPolicy": {
-		NewAPI:       "policy/v1beta1 PodSecurityPolicy",
-		DeprecatedIn: "v1.10",
-		RemovedIn:    "v1.16",
-	},
-	"apiextensions.k8s.io/v1beta1 CustomResourceDefinition": {
-		NewAPI:       "apiextensions.k8s.io/v1 CustomResourceDefinition",
-		DeprecatedIn: "v1.16",
-		RemovedIn:    "v1.19",
-	},
-	"extensions/v1beta1 Ingress": {
-		NewAPI:       "networking.k8s.io/v1beta1 Ingress",
-		DeprecatedIn: "v1.14",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRole": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRole",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleBinding": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBinding",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 ClusterRoleBindingList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBindingList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 Role": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 Role",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 RoleList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 RoleList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 RoleBinding": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBinding",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1alpha1 RoleBindingList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBindingList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta ClusterRole": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRole",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta1 ClusterRoleList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBinding",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta1 ClusterRoleBindingList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 ClusterRoleBindingList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta1 Role": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 Role",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta1 RoleList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 RoleList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta1 RoleBinding": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBinding",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-	"rbac.authorization.k8s.io/v1beta1 RoleBindingList": {
-		NewAPI:       "rbac.authorization.k8s.io/v1 RoleBindingList",
-		DeprecatedIn: "v1.17",
-		RemovedIn:    "v1.22",
-	},
-}
+const (
+	// This should be set based on the version of client-go being imported
+	k8sVersionMajor = 1
+	k8sVersionMinor = 19
+)
 
 // deprecatedAPIError indicates than an API is deprecated in Kubernetes
 type deprecatedAPIError struct {
-	Deprecated  string
-	Alternative string
-	In          string
-	Removed     string
+	Deprecated string
+	Message    string
 }
 
 func (e deprecatedAPIError) Error() string {
-	msg := fmt.Sprintf("the kind %q is deprecated as of %q and removed in %q", e.Deprecated, e.In, e.Removed)
-	if e.Alternative != "" {
-		msg += fmt.Sprintf(" in favor of %q", e.Alternative)
-	}
+	msg := e.Message
 	return msg
 }
 
 func validateNoDeprecations(resource *K8sYamlStruct) error {
-	gvk := fmt.Sprintf("%s %s", resource.APIVersion, resource.Kind)
-	if dep, ok := deprecatedAPIs[gvk]; ok {
-		return deprecatedAPIError{
-			Deprecated:  gvk,
-			Alternative: dep.NewAPI,
-			In:          dep.DeprecatedIn,
-			Removed:     dep.RemovedIn,
-		}
+	// resource is not a resource
+	if resource.APIVersion == "" {
+		return nil
 	}
-	return nil
+	if resource.Kind == "" {
+		return nil
+	}
+
+	runtimeObject, err := resourceToRuntimeObject(resource)
+	if err != nil {
+		return err
+	}
+	if !deprecation.IsDeprecated(runtimeObject, k8sVersionMajor, k8sVersionMinor) {
+		return nil
+	}
+	gvk := fmt.Sprintf("%s %s", resource.APIVersion, resource.Kind)
+	return deprecatedAPIError{
+		Deprecated: gvk,
+		Message:    deprecation.WarningMessage(runtimeObject),
+	}
+}
+
+func resourceToRuntimeObject(resource *K8sYamlStruct) (runtime.Object, error) {
+	scheme := runtime.NewScheme()
+	kscheme.AddToScheme(scheme)
+
+	gvk := schema.FromAPIVersionAndKind(resource.APIVersion, resource.Kind)
+	out, err := scheme.New(gvk)
+	if err != nil {
+		return nil, err
+	}
+	out.GetObjectKind().SetGroupVersionKind(gvk)
+	return out, nil
 }

--- a/pkg/lint/rules/deprecations.go
+++ b/pkg/lint/rules/deprecations.go
@@ -26,7 +26,8 @@ import (
 )
 
 const (
-	// This should be set based on the version of client-go being imported
+	// This should be set in the Makefile based on the version of client-go being imported.
+	// These constants will be overwritten with LDFLAGS
 	k8sVersionMajor = 1
 	k8sVersionMinor = 19
 )

--- a/pkg/lint/rules/deprecations.go
+++ b/pkg/lint/rules/deprecations.go
@@ -44,7 +44,7 @@ func (e deprecatedAPIError) Error() string {
 }
 
 func validateNoDeprecations(resource *K8sYamlStruct) error {
-	// resource is not a resource
+	// if `resource` does not have an APIVersion or Kind, we cannot test it for deprecation
 	if resource.APIVersion == "" {
 		return nil
 	}

--- a/pkg/lint/rules/deprecations.go
+++ b/pkg/lint/rules/deprecations.go
@@ -53,6 +53,10 @@ func validateNoDeprecations(resource *K8sYamlStruct) error {
 
 	runtimeObject, err := resourceToRuntimeObject(resource)
 	if err != nil {
+		// do not error for non-kubernetes resources
+		if runtime.IsNotRegisteredError(err) {
+			return nil
+		}
 		return err
 	}
 	if !deprecation.IsDeprecated(runtimeObject, k8sVersionMajor, k8sVersionMinor) {

--- a/pkg/lint/rules/deprecations_test.go
+++ b/pkg/lint/rules/deprecations_test.go
@@ -29,7 +29,7 @@ func TestValidateNoDeprecations(t *testing.T) {
 	}
 	depErr := err.(deprecatedAPIError)
 	if depErr.Message == "" {
-		t.Fatal("Expected error message to be non-blank")
+		t.Fatalf("Expected error message to be non-blank: %v", err)
 	}
 
 	if err := validateNoDeprecations(&K8sYamlStruct{

--- a/pkg/lint/rules/deprecations_test.go
+++ b/pkg/lint/rules/deprecations_test.go
@@ -27,10 +27,9 @@ func TestValidateNoDeprecations(t *testing.T) {
 	if err == nil {
 		t.Fatal("Expected deprecated extension to be flagged")
 	}
-
 	depErr := err.(deprecatedAPIError)
-	if depErr.Alternative != "apps/v1 Deployment" {
-		t.Errorf("Expected %q to be replaced by %q", depErr.Deprecated, depErr.Alternative)
+	if depErr.Message == "" {
+		t.Fatal("Expected error message to be non-blank")
 	}
 
 	if err := validateNoDeprecations(&K8sYamlStruct{

--- a/pkg/lint/rules/template.go
+++ b/pkg/lint/rules/template.go
@@ -137,8 +137,11 @@ func Templates(linter *support.Linter, values map[string]interface{}, namespace 
 				linter.RunLinterRule(support.ErrorSev, fpath, validateYamlContent(err))
 
 				if yamlStruct != nil {
-					linter.RunLinterRule(support.ErrorSev, fpath, validateMetadataName(yamlStruct))
-					linter.RunLinterRule(support.ErrorSev, fpath, validateNoDeprecations(yamlStruct))
+					// NOTE: set to warnings to allow users to support out-of-date kubernetes
+					// Refs https://github.com/helm/helm/issues/8596
+					linter.RunLinterRule(support.WarningSev, fpath, validateMetadataName(yamlStruct))
+					linter.RunLinterRule(support.WarningSev, fpath, validateNoDeprecations(yamlStruct))
+
 					linter.RunLinterRule(support.ErrorSev, fpath, validateMatchSelector(yamlStruct, renderedContent))
 				}
 			}

